enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/NullListAcceptanceTest.scala

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.internal.cypher.acceptance
+
+import org.neo4j.cypher._
+import org.neo4j.internal.cypher.acceptance.CypherComparisonSupport.Versions.{V2_3, V3_2}
+import org.neo4j.internal.cypher.acceptance.CypherComparisonSupport._
+
+class NullListAcceptanceTest extends ExecutionEngineFunSuite with CypherComparisonSupport {
+
+  // Changed behaviour to comply with opencypher in 3.3
+  val nullInListConfigOld: TestConfiguration = TestConfiguration(V2_3 -> V3_2, Planners.all, Runtimes.all) + Configs.AllRulePlanners
+
+  // Comparison between lists and non-lists
+
+  test("equality between list and literal should return null") {
+    val query = "WITH [1, 2] AS l1, 'foo' AS l2 RETURN l1 = l2 AS res"
```

Comment:
Why is this null and not false?

Reply:
According to the CIP: "Lists are incomparable to any value that is not also a list." Incomparable seems to be used equivalent with "should evaluate to null"